nexus/mgs-updates/src/rot_updater.rs

```diff
@@ -300,6 +292,14 @@ impl SpComponentUpdateHelper for ReconfiguratorRotUpdater {
                 return Ok(PrecheckStatus::UpdateComplete);
             }
 
+            // If the active slot does not match the expected active slot, it is possible
+            // another update is happening. Bail out.
+            if expected_active_slot.slot() != active {
+                return Err(PrecheckError::WrongActiveSlot {
+                    expected: expected_active_slot.slot, found: *active
+                })
+            }
+
```

Comment:
The tests helped me find a bug! This precondition should be checked _after_ we've checked there is nothing to do and returned a `PrecheckStatus::UpdateComplete`. When an update is done, a new active slot is set, so this precondition error would always return after an update was completed instead of reporting the update had completed.

Reply:
We'll also need to check the transient boot selection and the pending persistent boot selection. Also, the active slot should match the persistent boot selection.
These checks can't be done, and the conflicts won't be seen, until Hubris PR [#2050](https://github.com/oxidecomputer/hubris/pull/2050) is merged.
Any deviation would probably indicate a previous failed update. An ignition power-cycle is the big hammer that can be used, but we don't want bugs where we are continually power-cycling equipment. So, RoT and SP resets should be considered if transient and pending persistent are != None. Active != Persistent also needs to be considered.